dags/pull_wys_new.py

```diff
@@ -0,0 +1,115 @@
+from airflow.utils.trigger_rule import TriggerRule
+from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+from airflow.hooks.postgres_hook import PostgresHook
+
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from googleapiclient.discovery import build
+from dateutil.relativedelta import relativedelta
+
+try:
+    sys.path.append('/etc/airflow/data_scripts/wys/api/python/')
+    from wys_api import api_main
+    from wys_google_sheet import read_masterlist
+except:
+    raise ImportError("Cannot import functions to pull watch your speed data")
+
+
+SLACK_CONN_ID = 'slack'
+def task_fail_slack_alert(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+
+    if context.get('task_instance').task_id == 't1':
+        task_msg = """:cat_shocked: The Task {task} in Pull WYS dag failed, 
+			<@UHJA7GHQV> please check.""".format(
+            task=context.get('task_instance').task_id,)
+    
+    elif context.get('task_instance').task_id == 'check_wys_mobile_dupes_id':
+        task_msg = """:cat_shocked: There are dupes in `wys.mobile_sign_installations_dupes`. 
+			<@UHJA7GHQV> please check.""".format(
+            task=context.get('task_instance').task_id,)
+    
+    # else other msg for task2
+    else:
+        task_msg = """:blob_fail: The Task {task} in Pull WYS dag failed, 
+			<@UHJA7GHQV> please check.""".format(
+            task=context.get('task_instance').task_id,)    
+        
+    # this adds the error log url at the end of the msg
+    slack_msg = task_msg + """ (<{log_url}|log>)""".format(
+            log_url=context.get('task_instance').log_url,)
+    failed_alert = SlackWebhookOperator(
+        task_id='slack_test',
+        http_conn_id='slack',
+        webhook_token=slack_webhook_token,
+        message=slack_msg,
+        username='airflow',
+        )
+    return failed_alert.execute(context=context)
+
+#to get credentials to access google sheets
+wys_api_hook = GoogleCloudBaseHook('vz_api_google')
+cred = wys_api_hook._get_credentials()
+service = build('sheets', 'v4', credentials=cred, cache_discovery=False)
+
+#to connect to pgadmin bot
+wys_postgres = PostgresHook("wys_bot")
+connection = BaseHook.get_connection('wys_api_key')
+api_key = connection.password
+
+default_args = {'owner':'rdumas',
+                'depends_on_past':False,
+                'start_date': datetime(2020, 4, 1),
+                'email': ['raphael.dumas@toronto.ca'],
+                'email_on_failure': False,
+                 'email_on_success': False,
+                 'retries': 0,
+                 'retry_delay': timedelta(minutes=5),
+                 'on_failure_callback': task_fail_slack_alert
+                }
+
+dag = DAG('pull_wys_new',default_args=default_args, schedule_interval='0 15 * * *')
+# Run at 3 PM local time every day
+
+with wys_postgres.get_conn() as con:
+    t1 = PythonOperator(
```

Comment:
In general I think task variables should be the same as their `task_id` to make it easier to differentiate logic further down.